### DISCUSSION TITLE
GitHub Actions workflows stuck

### DISCUSSION THREAD

**Discussion by sdarwin** [community/community#64506] [Actions]
### Select Topic Area

Bug

### Body

Hi,

We are querying the github api to discover the status of Actions workflows and make programmatic decisions based on the result.

```
gh api repos/organization/repo/actions/runs?status=
```

Some workflows runs have gotten stuck for weeks or months:

https://github.com/boostorg/multiprecision/actions/runs/3253614037	# queued for 10 months

https://github.com/boostorg/math/actions/runs/4689052872		# in_progress 4 months

https://github.com/boostorg/lambda2/actions/runs/4707745899		# in_progress 4 months

Maybe this could be investigated, so future workflows don't also get the same problem. If possible, fix a bug. However, it might be difficult to discover the reason. In that case, perhaps add timeouts in GitHub Actions to end such stray workflows after a period of time.

**Comment by arafajs** [comment]
It is difficult to discover the reason why these workflows are getting stuck, then it might be a good idea to add timeouts to GitHub Actions. This would ensure that stray workflows are ended after a period of time.

In the meantime, you can try the following to troubleshoot the issue:

- Check the logs for the stuck workflows to see if there are any errors.
- Try running the workflows manually to see if they work.
- Try changing the settings for the workflows to see if that helps.

**Comment by sdarwin** [comment]
@arafajs  I believe the problem was transient, since subsequent runs were able to finish.  The issues happened a few months ago.  Since then, workflows have succeeded.   

The issue is not to debug these particular workflow files, since they are working now, but to point out GitHub seems to think these are still **in_progress** or **queued**.
